src/include/kdb.h.in

```diff
@@ -134,80 +125,62 @@ int kdbSet (KDB *handle, KeySet *returned,
  **************************************/
 
 /* Basic Methods */
-Key *keyNew (const char *keyname, ...) ELEKTRA_SENTINEL;
-Key *keyVNew (const char *keyname, va_list ap);
-
-Key * keyCopy (Key *dest, const Key *source, elektraCopyFlags flags);
-
-int keyClear (Key *key);
-int keyDel (Key *key);
-
-uint16_t keyIncRef (Key *key);
-uint16_t keyDecRef (Key *key);
-uint16_t keyGetRef (const Key *key);
+ElektraKey * elektraKeyNew (const char * keyname, ...) ELEKTRA_SENTINEL;
+ElektraKey * elektraKeyVNew (const char * keyname, va_list ap);
 
-/* Meta Info */
-int keyRewindMeta (Key *key);
-const Key *keyNextMeta (Key *key);
-const Key *keyCurrentMeta (const Key *key);
+ElektraKey * elektraKeyCopy (ElektraKey * dest, const ElektraKey * source, elektraCopyFlags flags);
 
-int keyCopyMeta (Key *dest, const Key *source, const char *metaName);
-int keyCopyAllMeta (Key *dest, const Key *source);
+int elektraKeyClear (ElektraKey * key);
+int elektraKeyDel (ElektraKey * key);
 
-const Key *keyGetMeta (const Key *key, const char* metaName);
-ssize_t    keySetMeta (Key *key, const char* metaName,
-	const char *newMetaString);
-KeySet * keyMeta (Key * key);
+uint16_t elektraKeyIncRef (ElektraKey * key);
+uint16_t elektraKeyDecRef (ElektraKey * key);
+uint16_t elektraKeyGetRef (const ElektraKey * key);
 
-/* Methods for Making Tests */
-int keyCmp (const Key *k1, const Key *k2);
+/* Metadata */
+ElektraKeySet * elektraKeyMeta (ElektraKey * key);
+int elektraKeySetMeta (ElektraKey * key, ElektraKeySet * meta);
```

Comment:
One thing I noticed while reading this, is that it might get misconstrued as KeySet Meta

Reply:
Yes, I noticed that too. That's why I called all the keyset functions `elektraKeyset*`. Probably, the struct should also be called `ElektraKeyset` to avoid confusion.

Reply:
Yes, I just noticed as I was updating the Keyset functions. At least alleviates the issues, renaming the struct is a good idea as well. I tried to think of better names, but this would just make everything inconsistent again I guess.

Reply:
Yes, if we prefix with elektra we also need to rename Keyset. Alternative would be KS, resembling KDB.